lib/filewatch/watch.rb

```diff
@@ -62,62 +45,66 @@ def initialize(opts={})
       end
       @watching = []
       @exclude = []
-      @files = Hash.new { |h, k| h[k] = WatchedFile.new(k, false, false) }
-      @unwatched = Hash.new
+      @files = Hash.new
       # we need to be threadsafe about the mutation
       # of the above 2 ivars because the public
-      # methods each, discover, watch and unwatch
+      # methods each, discover and watch
       # can be called from different threads.
       @lock = Mutex.new
       # we need to be threadsafe about the quit mutation
       @quit = false
       @quit_lock = Mutex.new
+      @max_active = ENV.fetch("FILEWATCH_MAX_OPEN_FILES", 4095).to_i
+      @max_warn_msg = "Reached maximum (#{@max_active}) open files"
+      @lastwarn_max_files = 0
     end # def initialize
 
     public
+
+    def max_open_files=(value)
+      val = value.to_i
+      return if value.nil? || val <= 0
+      @max_warn_msg = "Reached maximum (#{val}) open files"
```

Comment:
Good idea. I am adding this to the file input already:
```ruby
  # What is the maximum number of file_handles that this input consumes
  # at any one time. Use close_older to close some files if you need to
  # process more files than this number. This should not be set to the 
  # maximum the OS can do because file handles are needed for other
  # LS plugins and OS processes.
  # The default of 4095 is set in filewatch.
  config :max_open_files, :validate => :number
```